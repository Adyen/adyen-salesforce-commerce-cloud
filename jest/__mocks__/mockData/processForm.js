export const getPaymentForm = jest.fn(() => ({
  paymentMethod: {
    value: 'mockedPaymentMethod',
  },
  creditCardFields: {
    cardType: {
      value: 'mockedCardType',
    },
    cardNumber: {
      value: 'mockedCardNumber',
    },
    saveCard: {
      checked: true,
    },
  },
  adyenPaymentFields: {
    adyenFingerprint: {
      value:
        'ryEGX8eZpJ0030000000000000KZbIQj6kzs0089146776cVB94iKzBGQGbvPiVrHq5S16Goh5Mk004ivbSuYdG0R00000YVxEr00000cru9sAxTR5wwQsLYbcA8:40',
    },
    adyenStateData: {
      value: 'mockedStateData',
    },
    adyenPartialPaymentsOrder: {
        value: 'mockedOrder'
    }
  },
}));

export const getPaymentInstruments = jest.fn((UUID = 'mocked_uuid') => ([{
  UUID,
  creditCardNumber: 'mocked_cc_number',
  creditCardType: 'mocked_cc_type',
  creditCardExpirationMonth: 'mocked_cc_exp_month',
  creditCardExpirationYear: 'mocked_cc_exp_year',
  creditCardToken: 'mocked_raw_cc_token ',
  getCreditCardToken: () => 'mocked_id'
}]));
