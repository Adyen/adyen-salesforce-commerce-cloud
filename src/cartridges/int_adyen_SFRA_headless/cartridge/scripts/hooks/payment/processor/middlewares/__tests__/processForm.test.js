import { getPaymentForm, getPaymentInstruments } from 'mockData/processForm';

/* eslint-disable global-require */
let processForm;
let req;
let adyenHelper;

beforeEach(() => {
  processForm = require('../processForm');
  jest.clearAllMocks();
  req = {
    form: {
      adyenPaymentMethod: 'mockedPaymentMethod',
      adyenIssuerName: 'mocked_issuer_name',
      brandCode: 'scheme',
    },
    currentCustomer: { raw: {} },
  };
  adyenHelper = require('*/cartridge/scripts/util/adyenHelper');
});

describe('processForm', () => {
  it('should return error when credit card validation fails', () => {
    const processFormOutput = processForm(req, getPaymentForm(), {});
    expect(processFormOutput).toMatchSnapshot();
  });

  it('should return viewData', () => {
    req.form.storedPaymentUUID = 'mockedUUID';
    const processFormOutput = processForm(req, getPaymentForm(), {});
    expect(processFormOutput).toMatchSnapshot();
  });

  it('should return viewData when authenticated and registered', () => {
    const paymentForm = getPaymentForm();
    const uuid = 'mocked_id';
    const paymentInstrument = getPaymentInstruments(uuid);
    paymentForm.creditCardFields.selectedCardID = { value: uuid };
    paymentForm.adyenPaymentFields.adyenStateData = {
      value: JSON.stringify({
        paymentMethod: {
          storedPaymentMethodId: 'mocked_id',
        }
      }),
    }
    req.form.brandCode = 'not_scheme';
    req.form.securityCode = 'mocked_security_code';
    req.currentCustomer.raw = { authenticated: true, registered: true };
    adyenHelper.getCustomer.mockReturnValue({
      getProfile: () => {
        return {
          getWallet: () => {
            return {
              getPaymentInstruments: () => paymentInstrument
            }
          }
        }
      }
    })
    const processFormResult = processForm(req, paymentForm, {});
    expect(processFormResult).toMatchSnapshot();
  });
});
