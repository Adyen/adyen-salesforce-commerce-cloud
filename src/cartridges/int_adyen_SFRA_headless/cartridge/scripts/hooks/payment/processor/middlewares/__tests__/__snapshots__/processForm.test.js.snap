// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`processForm should return error when credit card validation fails 1`] = `
{
  "error": true,
  "fieldErrors": {
    "creditCardErrors": "mockedCreditCardErrors",
  },
}
`;

exports[`processForm should return viewData 1`] = `
{
  "error": false,
  "viewData": {
    "paymentInformation": {
      "adyenIssuerName": "mocked_issuer_name",
      "adyenPaymentMethod": "mockedPaymentMethod",
      "cardNumber": "mockedCardNumber",
      "cardType": "mockedCardType",
      "isCreditCard": true,
      "partialPaymentsOrder": "mockedOrder",
      "stateData": "mockedStateData",
    },
    "paymentMethod": {
      "htmlName": "mockedPaymentMethod",
      "value": "mockedPaymentMethod",
    },
    "saveCard": true,
  },
}
`;

exports[`processForm should return viewData when authenticated and registered 1`] = `
{
  "error": false,
  "viewData": {
    "paymentInformation": {
      "adyenIssuerName": "mocked_issuer_name",
      "adyenPaymentMethod": "mockedPaymentMethod",
      "cardNumber": "mocked_cc_number",
      "cardType": "mocked_cc_type",
      "creditCardToken": "mocked_raw_cc_token ",
      "expirationMonth": "mocked_cc_exp_month",
      "expirationYear": "mocked_cc_exp_year",
      "isCreditCard": false,
      "partialPaymentsOrder": "mockedOrder",
      "securityCode": "mocked_security_code",
      "stateData": "{"paymentMethod":{"storedPaymentMethodId":"mocked_id"}}",
    },
    "paymentMethod": {
      "htmlName": "mockedPaymentMethod",
      "value": "mockedPaymentMethod",
    },
    "saveCard": true,
  },
}
`;
